### DISCUSSION TITLE
How fetchOptions from StoreManager works ?

### DISCUSSION THREAD

**Discussion by AntoineGGG** [GrapesJS/grapesjs#4277] [Q&A]
Hey guys, 
First thanks @artf  for your works !!

I'm in trouble to manage the StoreManager fetchOptions.
I would like to use like this :

fetchOptions: id ? {method: 'PATCH'} : {method:"POST"}

But nothing works for now.

I tried to write fetchOptions: {method:"PATCH"} but the request still be a POST method. 
Any advise to make it works ?

Thanks a lot..

**Comment by artf** [comment]
Did you try with something like this
```js
fetchOptions: opts => {
 return opts.method === 'post' ?  { method: 'PATCH' } : {}
},
```

**Reply by AntoineGGG** [reply]
Thanks for your reply, I just tried this and it's working, but I writte it like this :
`fetchOptions: () => {
    return id ? { method: "PATCH" } : { method: "POST" };
  },`
With this I have a problem cause the GET method is missing and the editor try to fetch content with a patch...
